content/300-guides/300-upgrade-guides/200-upgrading-versions/050-upgrading-to-prisma-3/150-referential-actions.mdx

```diff
@@ -0,0 +1,187 @@
+
+</Admonition>
+
+Make sure you are happy with every case of `onDelete: Cascade` in your schema. If not, either:
+
+- Modify your Prisma schema and `db push` or `dev migrate` to change the database*or*
+- Manually update the underlying database if you use an introspection-only workflow
+
+The following example would result in a cascading delete, if the `User` is deleted then all of their `Post`'s will be deleted too.
+
+#### A blog schema example
+
+```prisma highlight=4;add
+model Post {
+  id       Int    @id @default(autoincrement())
+  title    String
+  author   User   @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+### Using Migration
+
+When running a [Migration](../../../../concepts/components/prisma-migrate) (or the [`prisma db push`](../../../prisma-migrate/db-push) command) the [new defaults](../../../../concepts/components/prisma-schema/relations/referential-actions#referential-action-defaults) will be applied to your database.
+
+<Admonition type="info">
+
+Unlike when you run an Introspect for the first time, the new referential actions clause and property, will **not** automatically be addded to your prisma schema by the Prisma VSCode extension.
+You will have to manually add them if you wish to use anything other than the new defaults.
+
+</Admonition>
+
+Explicitly defining referential actions in your Prisma schema is optional. If you do not explicitly define a referential action for a relation, Prisma uses the [new defaults](../../../../concepts/components/prisma-schema/relations/referential-actions#referential-action-defaults).
+
+Note that referential actions can be added on a case by case basis. This means that you can add them to one single relation and leave the rest set to the defaults by not manually specifying anything.
+
+### Checking for errors
+
+**Before** upgrading to 2.26.0, Prisma prevented the deletion of records while using `delete()` or `deleteMany()` to preserve referential integrity. A custom runtime error would be thrown by the Prisma Client with the error code `P2014`.
+
+**After** upgrading, Prisma no longer performs runtime checks. You can instead specify a custom referential action to preserve the referential integrity between relations.
+
+When you use [`NoAction`](#noaction) or [`Restrict`](#restrict) to prevent the deletion of records, the error messages will be different post 2.26.0 compared to pre 2.26.0. This is because they are now triggered by the database and **not** the Prisma Client. The new error code that can be expected is `P2003`.
```

Comment:
```suggestion
When you use [`NoAction`](#noaction) or [`Restrict`](#restrict) to prevent the deletion of records, the error messages will be different in versions 3.0.1 and above (or 2.26.0 with the `referentialActions` feature flag enabled) compared to versions prior to that. This is because they are now triggered by the database and **not** the Prisma Client. The new error code that can be expected is `P2003`, so you should check your code to make adjustments accordingly.
```

Reply:
Same thing, not entirely sure if we feel that versioning detail matters enough, but I'd err on the side of using 3.0.1 wherever possible since we're in the upgrade guide from 2 to 3